### ISSUE TITLE
Failed to upgrade the helm chart

### ISSUE THREAD

**Issue by RobertIndie** [apache/streampipes#1546]
### Apache StreamPipes version

dev (current development state)

### Affected StreamPipes components

Backend, Connect, Processing Elements, UI

### What happened?

When running `helm upgrade streampipes ./` to upgrade the helm chart, it will throw errors:
```
Error: UPGRADE FAILED: cannot patch "backend-pvc" with kind PersistentVolumeClaim: PersistentVolumeClaim "backend-pvc" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "50Mi", Format: "BinarySI"}}},
  	VolumeName:       "pvc-a85d92d5-ab2f-4500-9d22-326629e3365b",
- 	StorageClassName: &"hostpath",
+ 	StorageClassName: nil,
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }
 && cannot patch "consul-pvc" with kind PersistentVolumeClaim: PersistentVolumeClaim "consul-pvc" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "20M", Format: "DecimalSI"}}},
  	VolumeName:       "pvc-89730b87-f414-48a9-b842-58a677f2ddb6",
- 	StorageClassName: &"hostpath",
+ 	StorageClassName: nil,
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }
 && cannot patch "couchdb-pvc" with kind PersistentVolumeClaim: PersistentVolumeClaim "couchdb-pvc" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "50M", Format: "DecimalSI"}}},
  	VolumeName:       "pvc-1f227581-aa7c-4ed4-a30f-3c2abc302118",
- 	StorageClassName: &"hostpath",
+ 	StorageClassName: nil,
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }
 && cannot patch "influxdb-pvc" with kind PersistentVolumeClaim: PersistentVolumeClaim "influxdb-pvc" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "250Mi", Format: "BinarySI"}}},
  	VolumeName:       "pvc-e1f7a48e-5c10-4d57-90c5-6a95d5eab867",
- 	StorageClassName: &"hostpath",
+ 	StorageClassName: nil,
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }
 && cannot patch "influxdb2-pvc" with kind PersistentVolumeClaim: PersistentVolumeClaim "influxdb2-pvc" is invalid: spec: Forbidden: spec is immutable after creation except resources.requests for bound claims
  core.PersistentVolumeClaimSpec{
  	... // 2 identical fields
  	Resources:        {Requests: {s"storage": {i: {...}, s: "250Mi", Format: "BinarySI"}}},
  	VolumeName:       "pvc-776895f1-3137-4853-b02a-22eae07f7cc7",
- 	StorageClassName: &"hostpath",
+ 	StorageClassName: nil,
  	VolumeMode:       &"Filesystem",
  	DataSource:       nil,
  	DataSourceRef:    nil,
  }
```

### How to reproduce?

1. Deploy the streampipes according to https://github.com/apache/streampipes/blob/dev/installer/k8s/README.md
2. Upgrade the helm chart by executing `helm upgrade streampipes ./`.

### Are you willing to submit a PR?

None

**Comment by bossenti** [comment]
@smlabt maybe that's something where you can jump in ☺️

**Comment by luoluoyuyu** [comment]
hi @RobertIndie 
I think the reason for the error is that the value of StorageClassName is the empty string, so pv uses hostpath by default, we should change the value of StorageClassName to hostpath.

**Comment by RobertIndie** [comment]
@luoluoyuyu Thanks for investigating. I assign this issue to you since you are already working on this issue.

**Comment by bossenti** [comment]
resolved by #1775
